code/game/gamemodes/messiah/miracles.dm

```
@@ -0,0 +1,97 @@
+			kotime = rand(10,60)
+			spawn(kotime)
+				target.apply_effect(rand(5,20), PARALYZE)
+
+/obj/effect/proc_holder/spell/targeted/jesus_revive
+	name = "Miracle: Ressurection!"
+	desc = "After a short channeling period, this miracle brings a target back from the dead."
+	clothes_req = 0
+	max_targets = 1
+	range = 1
+	panel = "Miracles"
+	charge_max = 400
+	invocation_type = "none"
+
+/obj/effect/proc_holder/spell/targeted/jesus_revive/cast(list/targets,mob/user = usr)
+
+	for(var/mob/living/carbon/human/target in targets)
+		if(target.stat == DEAD)
+			target.notify_ghost_cloning("[user] the Messiah is attempting to give you life. Re-enter your corpse if you want to be revived!")
+			if(do_mob(user,target,80))
+				target.revive()
+		else
+			user << "Ressurection only works on the dead!"
+
+/obj/effect/proc_holder/spell/targeted/jesus_revive/cure
+	name = "Miracle: Heal The Sick!"
+	desc = "With the power of this miracle, Jesus can return any living being to full health and will remove any disease or mutation affecting the target."
+	range = 7
+	charge_max = 200
+
+/obj/effect/proc_holder/spell/targeted/jesus_revive/cure/cast(list/targets,mob/user = usr)
+
+	for(var/mob/living/carbon/human/target in targets)
+		if(target.stat != DEAD)
+			target.apply_effect(8, PARALYZE)
+			target.revive()
+		else
+			user << "Use Ressurection to bring somebody back to life!"
+
+/obj/effect/proc_holder/spell/targeted/jesus_deconvert
+	name = "Miracle: Repent for your Sins!"
+	desc = "After channeling for two minutes, this miracle will show any sinner the righteous path, removing their antagonist status."
+	clothes_req = 0
+	max_targets = 1
+	range = 1
+	panel = "Miracles"
+	charge_max = 6000
+	invocation_type = "none"
+
+/obj/effect/proc_holder/spell/targeted/jesus_deconvert/cast(list/targets,mob/user = usr)
+
+	for(var/mob/living/carbon/human/target in targets)
+		if(target.stat != DEAD || target.mind)
+			var/is_antag = target.mind.special_role
+			if(do_mob(user,target,400))
```

Won't this just silently fail if you use it on a good guy?